src/main/java/seedu/address/logic/commands/task/AddTaskCommand.java

```diff
@@ -37,38 +35,41 @@
     public static final String MESSAGE_ADD_TASK_SUCCESS = "New Task added to Room %1$d. \nDescription: %2$s";
 
     private final Task taskToAdd;
-    private final Index roomIndex;
+    private final int roomNumber;
 
     /**
      * Creates an AddTaskCommand to add the specified {@code Task} to a {@code Room}
-     * with the {@code roomIndex}.
+     * with the {@code roomNumber}.
      */
-    public AddTaskCommand(Task task, Index roomIndex) {
-        requireAllNonNull(task, roomIndex);
+    public AddTaskCommand(Task task, int roomNumber) {
+        requireNonNull(task);
+        assert roomNumber > 0 : "Room number should be greater than 0.";
         taskToAdd = task;
-        this.roomIndex = roomIndex;
+        this.roomNumber = roomNumber;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Room> rooms = model.getRoomList();
-
-        if (roomIndex.getZeroBased() >= rooms.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_ROOM_INDEX);
+        if (roomNumber < 0) {
```

Comment:
Yup that's true. Good observation here 😄

That said, assertions may not necessarily be enabled, especially when an application goes into production.

The `roomNumber < 0` check is to prevent the application from crashing if this method somehow gets called when assertions are off.